lib/matplotlib/figure.py

```diff
@@ -3125,32 +3138,32 @@ def clear(self, keep_observers=False):
     def draw(self, renderer):
         # docstring inherited
 
-        # draw the figure bounding box, perhaps none for white figure
-        if not self.get_visible():
-            return
+        with self._render_lock:
+            if not self.get_visible():
+                return
```

Comment:
Leave this outside the lock to save acquiring the lock and releasing right away?